### PR TITLE
Fix threadpool issue #11875

### PR DESCRIPTION
**Pull request by alehander92**

### CODE REVIEW

**Comment by alehander42** [comment]
can't understand which test fails: only on travis

File: lib/pure/concurrency/threadpool.nim [hunk 1]
```diff
@@ -321,8 +321,8 @@ type
 var
   currentPoolSize: int
   maxPoolSize = MaxThreadPoolSize
-  minPoolSize = 4
-  gSomeReady: Semaphore
+  minPoolSize = 1
+  gSomeReady : Semaphore
```
Comment (Araq) [inline lib/pure/concurrency/threadpool.nim]:
```suggestion
  gSomeReady: Semaphore
```

File: lib/pure/concurrency/threadpool.nim [hunk 2]
```diff
@@ -590,11 +590,14 @@ proc sync*() =
...
-      allReady = allReady and workersData[i].ready
-    if allReady: break
+    var count = 0
+    
+    for worker in workersData:
+      if worker.ready:
+        count += 1
+      if count >= currentPoolSize:
+        return
+
```
Comment (Araq) [inline lib/pure/concurrency/threadpool.nim]:
```suggestion
    for worker in workersData:
      if worker.ready:
        count += 1
      if count >= currentPoolSize:
        return
```

**Comment by Araq** [comment]
Malebolgia works better anyway.

**Pull request closed without merging**